src/test/java/org/jabref/gui/externalfiles/UnlinkedFilesDialogViewModelTest.java

```diff
@@ -0,0 +1,119 @@
+package org.jabref.gui.externalfiles;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javafx.beans.property.SimpleListProperty;
+import javafx.collections.FXCollections;
+import javafx.scene.control.TreeItem;
+
+import org.jabref.gui.StateManager;
+import org.jabref.gui.preferences.GuiPreferences;
+import org.jabref.gui.util.FileNodeViewModel;
+import org.jabref.logic.FilePreferences;
+import org.jabref.logic.util.TaskExecutor;
+import org.jabref.model.database.BibDatabaseContext;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class UnlinkedFilesDialogViewModelTest {
+    @TempDir
+    Path tempDir;
+    @TempDir
+    Path subDir;
+    @TempDir
+    Path file1;
+    @TempDir
+    Path file2;
+    @Mock
+    private TaskExecutor taskExecutor;
+    @Mock
+    private GuiPreferences guiPreferences;
+    @Mock
+    private StateManager stateManager;
+    @Mock
+    private BibDatabaseContext bibDatabaseContext;
+
+
+    private UnlinkedFilesDialogViewModel viewModel;
+    @BeforeEach
+    public void setUp() throws Exception {
+        MockitoAnnotations.openMocks(this);
```

Comment:
shouldn't this be in a try-with-resources block?

Reply:
No, because it is `@BeforeEach`. In a try-with-resources block, the mocks are closed **before** the test case runs. But we need the mocks in the test case: We declared the mocks as class variable because we want to use it at `@Test` methods. If we initialize the mocks in `setUp()`, close it at the end of `setUp()`, we would need to initilaize them again in **each** test method.

If we initialize the mocks in each test method, that would be more "clean". However, the resources are freed after the tests. This is not a long running process (taking much RAM forever), but a short one. Thus, I think, the code readability (mocks initialized once instead of in each test) outweighs the code cleaniness (try-with-resources).